File: implementations/java/compression/src/utils/fractal/mapping/FractalMapping.java
package implementations.java.compression.src.utils.fractal.mapping;

import implementations.java.compression.src.utils.fractal.transformation.Transformation;
import implementations.java.compression.src.utils.fractal.transformation.TransformationFactory;
import implementations.java.compression.src.utils.fractal.transformation.TransformationType;

public record FractalMapping(int rangeX, int rangeY, int domainX, int domainY, float s, float o, TransformationType t) {

    public Transformation transformation() {
        return TransformationFactory.of(t);
    }

}

File: implementations/cpp/fractal/mapping/fractal-mapping.cpp
#include "fractal-mapping.hpp"

FractalMapping::FractalMapping() : range_x(0), range_y(0), domain_x(0), domain_y(0), s(0), o(0) {}

FractalMapping::FractalMapping(int range_x, int range_y, int domain_x, int domain_y, float s, float o)
    : range_x(range_x), range_y(range_y), domain_x(domain_x), domain_y(domain_y), s(s), o(o) {}
